### PR TITLE
Fix reported bug where no-call genotypes with no reads get genotype

### LINKED ISSUE #5666: CalculateGenotypePosteriors shouldn't modify GTs with no data
A user reported that CalculateGenotypePosteriors generated a bunch of "new" variants (https://gatkforums.broadinstitute.org/gatk/discussion/comment/56226).  What happened was that no-calls that look like `GT:AD:DP:GQ:PL ./.:0,0:0:.:0,0,0` became `GT:AD:DP:GQ:PL:PP 0/1:0,0:0:,:30,0,30` or similar when a high AF prior was applied.  The expected behavior is for PPs to not be applied and the GT to remain no-call.

### PR DESCRIPTION
**Pull request by ldgauthier**
posterior probabilities and calls

Closes #5666

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/PosteriorProbabilitiesUtils.java [hunk 1]
```diff
@@ -187,7 +187,26 @@ else if (a.isSymbolic()) {   //use the greater of the priors for non-ref
...
+
+    /**
+     * return true if the genotype contains real data for likelihoods, not just zero placeholder
+     */
+    private static boolean hasRealLikelihoods(Genotype genotype) {
+        if (!genotype.hasLikelihoods()) {
+            return false;
+        }
+        if (genotype.hasDP() && genotype.getDP() == 0) {
+            int[] pls = genotype.getPL();
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/PosteriorProbabilitiesUtils.java]: How about `return MathUtils.max(genotype.getPL()) > 0`?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/PosteriorProbabilitiesUtilsUnitTest.java [hunk 2]
```diff
@@ -391,6 +391,18 @@ public void testDifferentRefAllelesAndNonRefCounts() {
         Assert.assertEquals(arraysEq(expectedPPs, _mleparse((List<Integer>)test1result.getGenotype(0).getAnyAttribute(GATKVCFConstants.PHRED_SCALED_POSTERIORS_KEY))), "");
     }
 
+    @Test
+    public void testGenotypesWithNoDataDoNotChange() {
+        final GenotypeBuilder gb = new GenotypeBuilder("s1");
+        gb.alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL)).DP(0).AD(new int[]{0,0}).GQ(0).PL(new int[]{0,0,0});
```
Comment (davidbenjamin) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/PosteriorProbabilitiesUtilsUnitTest.java]: How about chaining `.alleles etc` with the previous line?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/PosteriorProbabilitiesUtilsUnitTest.java [hunk 3]
```diff
@@ -391,6 +391,18 @@ public void testDifferentRefAllelesAndNonRefCounts() {
...
+        final VariantContext vc = makeVC("noCall", Arrays.asList(Allele.create("A",true), Allele.create("T")), gb.make());
```
Comment (davidbenjamin) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/PosteriorProbabilitiesUtilsUnitTest.java]: `Allele.create("A", true)` can be replaced with the pre-defined `Allele.REF_A`.

**Comment by ldgauthier** [comment]
Back to you @davidbenjamin

**Comment by davidbenjamin** [comment]
Good to merge! :thumbsup:

**Pull request merged**